unicorn/include/cocaine/idl/unicorn.hpp

```diff
@@ -0,0 +1,267 @@
+        **/
+        typedef boost::mpl::list<
+            path_t,
+            value_t
+        > argument_type;
+
+        /**
+        * true if node was created. Error on any kind of error
+        */
+        typedef option_of<
+            bool
+        >::tag upstream_type;
+
+        typedef unicorn_final_tag dispatch_type;
+    };
+
+    struct put {
+        typedef unicorn_tag tag;
+
+        static const char* alias() {
+            return "put";
+        }
+
+        /**
+        *  put command implements cas behaviour. It accepts:
+        *
+        * path_t - path to change.
+        * value_t - value to write in path
+        * version_t - version to compare with. If version in zk do not match - error will be returned.
+        *   -1 indicates that version check is not performed (forced put).
+        **/
+        typedef boost::mpl::list<
+            path_t,
+            value_t,
+            version_t
+        > argument_type;
+
+        /**
+        * Return value is current value in ZK.
+        */
+        typedef option_of<
+            bool,
+            versioned_value_t
+        >::tag upstream_type;
+
+        typedef unicorn_final_tag dispatch_type;
+    };
+
+    struct subscribe {
+        typedef unicorn_tag tag;
+
+        static const char* alias() {
+            return "subscribe";
+        }
+
+        /**
+        * subscribe for updates on path. Will send last update which version is greater than specified.
+        */
+        typedef boost::mpl::list<
+            path_t
+        > argument_type;
+
+        /**
+        * current version in ZK
+        */
+        typedef stream_of<
+            versioned_value_t
+        >::tag upstream_type;
+
+        typedef unicorn_final_tag dispatch_type;
+    };
+
+    struct del {
+        typedef unicorn_tag tag;
+
+        static const char* alias() {
+            return "del";
+        }
+
+        /**
+        * delete node. Will only succeed if there are no child nodes.
+        */
+        typedef boost::mpl::list<
+            path_t,
+            version_t
+        > argument_type;
+
+        typedef option_of<
+            bool
+        >::tag upstream_type;
+
+        typedef unicorn_final_tag dispatch_type;
+    };
+
+    struct increment {
+        typedef unicorn_tag tag;
+
+        static const char* alias() {
+            return "increment";
+        }
+
+        /**
+        * increment node value in path by passed value.
+        * If either passed or stored value is not numeric will return error.
+        * If one of the values is float - result value will be float.
+        */
+        typedef boost::mpl::list<
+            path_t,
+            value_t
+        > argument_type;
+
+        /**
+        * return value after increment
+        */
+        typedef option_of <
+            versioned_value_t
+        >::tag upstream_type;
+
+        typedef unicorn_final_tag dispatch_type;
+    };
+
+    struct lsubscribe {
```

Comment:
Why so strange name `lsubscribe`? Also there aren't comments what's this event doing on invocation and what each parameter means.